homeassistant/components/recorder.py

```diff
@@ -194,6 +204,7 @@ def run(self):
         """Start processing events to save."""
         self._setup_connection()
         self._setup_run()
+        self._purge_old_data()
```

Comment:
As this will break your tests, you can just call `recorder._purge_old_data()` in your tests